### ISSUE TITLE
add support for properties with SortedSet and other collection datatype

### ISSUE THREAD

**Issue by paulovitor** [six2six/fixture-factory#60]
When execute fixture, following exception is displayed:

target = {java.lang.ClassCastException@1827}"java.lang.ClassCastException: Cannot cast java.util.HashSet to java.util.SortedSet"

```
public class Father {
    private SortedSet<Son> children = new TreeSet<>();
}
```

```
Fixture.of(Father.class).addTemplate("valid", new Rule() {
            {
                add("children", has(1).of(Son.class, "valid"));
            }
        });
```

Is it can add support for other collection beyond Set?

```
protected TransformerChain buildTransformerChain(Transformer transformer) {
        TransformerChain transformerChain = new TransformerChain(transformer);

        transformerChain.add(new CalendarTransformer());
        transformerChain.add(new SetTransformer());
        transformerChain.add(new PrimitiveTransformer());
        transformerChain.add(new WrapperTransformer());

        return transformerChain;
    }
```

Or get implementation class from Set interface, instead of create HashSet:

```
public <T> T transform(Object value, Class<T> type) {
        return type.cast(new HashSet((Collection) value));
    }
```

**Comment by MauricioJr** [comment]
@paulovitor I fixed it here f1d4dab. 
Any instance of Set should transformable . Let's wait code review and merge.

**Comment by paulovitor** [comment]
Ok Mauricio, thanks! ;)

**Comment by MauricioJr** [comment]
@ahirata @aparra @nykolaslima Hello guys, any chance to have it reviewed soon?

**Comment by maurcarvalho** [comment]
Fixed at https://github.com/six2six/fixture-factory/commit/f1d4dab135945f4de1ace6438f18f3a896100b99 and merged.
@ahirata or @paulovitor, could you please close this issue?
Thank you.

**Comment by paulovitor** [comment]
Fixed at [f1d4dab](https://github.com/six2six/fixture-factory/commit/f1d4dab135945f4de1ace6438f18f3a896100b99) and merged
